docs/developer-guide/cli-options.md

```diff
@@ -0,0 +1,120 @@
+# Command Line Options
+
+These instructions explain how to add new command-line options to the SIPNET executable. Three types of command-line 
+options are enabled: flags, integer-valued options, and string-valued options.
+
+## Background
+
+The **`Context` struct**, defined in `common/context.h`, holds the values of all command-line options.
+
+When adding a new command-line option, you will need to add a corresponding member to the `Context` struct that will hold the value of that option.
+
+## Important Notes
+
+**A note on naming:** 
+
+Each command-line option must have a corresponding entry in the `Context` struct. SIPNET uses a mapping system to 
+link the command-line option (e.g. `print_header`) to its `Context` member (in this case, `printHeader`).
+
+Some care must be taken to allow SIPNET to map between the option and the `Context` member that holds the options value. 
+The mapping is created using the `nameToKey` function in `common/context.c` that:
+- removes all non-alphanumeric characters.
+- converts alpha characters to lowercase. 
+
+For the mapping to work, the option name (e.g., `print_header`) and `Context` member name (in this case, `printHeader`) must generate the same key (e.g. `print_header` and `printHeader` both generate `printheader`).
```

Comment:
I changed this since #107 to be more informative
- previously it said: "`printheader` and `printHeader` both generate `printheader`"
- now (more usefully): "`print_header` and `printHeader` both generate `printheader`"